### DISCUSSION TITLE
ASF optimal against population or the Parento front, which is better?

### DISCUSSION THREAD

**Discussion by dinohsu1019** [anyoptimization/pymoo#515] [Q&A] [answered]
In my application, I use ASF optimal as the solution in a multi-objective optimization problem. I studied a lecture where ASF optimal is calculated against the population (res.X, res.F), not the Pareto Front (res.opt.get("X"), res.opt.get("F")). But my observation shows in the last generation, the Pareto Front (the parent or first front, with non-dominated solutions) size can be the same as or be less than the population size, in the former case, there's no difference, but in the latter case, is it better to calculate the ASF optimal against the Pareto Front only? Please shed some light, thanks in advance.

**Comment by blankjul** [comment]
Let us get the terminology right before starting.
Pareto-front is the known optimum in the objective space. For test problems, we can derive it mathematically; for real-world problems it is unknown. An algorithm finds a non-dominated set of solutions that we desire to be as close as possible to the Pareto-front.

Okay, all of that being said. ASF always returns exactly the same value no matter how many solutions you have. However, normalization will change it: https://pymoo.org/getting_started/part_3.html
If the ideal and nadir point is known to you, you can fix it during the run to calculate ASF in each iteration.

**Reply by dinohsu1019** [reply, marked as answer]
I do normalize objectives to calculate ASF for the result as well as for each iteration (so that I can see the optimal solution in each generation).

I realized later that result.F has the same number of solutions as result.opt.get("F"), which are the number of non-dominated solutions, as for the whole population, I can get them from result.pop.get("F"). Likewise, the same distinction can be seen for algorithm.opt.get("F") and algorithm.pop.get("F"), as well as algorithm.history[gen_i].opt.get("F") and algorithm.history[gen_i].pop.get("F"). In either case, ASF should be calculated against the "opt" counterpart, which is also the set of non-dominated solutions of the corresponding generation.

Thanks.
